src/backend/access/transam/xlog.c

```diff
@@ -8207,7 +8207,8 @@ StartupXLOG(void)
 	if (fast_promoted)
 		RequestCheckpoint(CHECKPOINT_FORCE);
 
-	*shmCleanupBackends = true;
+	 if (Gp_role == GP_ROLE_DISPATCH)
+		*shmCleanupBackends = true;
```

Comment:
do we need [this](https://github.com/arenadata/gpdb/commit/b04175666aefe49f7410ff010d84998d86d20d4c#diff-c1cb3ab2a19606390c1a7ed00ffe5a45531702ca5faf999d401c548f8951c65bR6294-R6296) change too?

Reply:
Do you mean to merge the whole patch "Try to terminate mpp backends during dtx recovery." or just this particular point of triggering restart? 
7.x has reworked restart procedure, I have doubts it's good idea to backport the whole patch.

Reply:
maybe, I don't remember why I didn't port this place.